### DISCUSSION TITLE
How to return a custom failure message from Event Listener service

### DISCUSSION THREAD

**Discussion by mlkdev** [pimcore/pimcore#12172] [Q&A / Support] [answered]
We have a simple service running to check a specific DataObject instance and if any tags have been applied to it before it gets published. We currently have the business logic working as expected, if a DataObject is published without tags, the service unpublishes the DataObject as expected, and we currently `return 0;` to display a "Saving failed." notice to the user.

My question is how do we provide a more informative message to the user so they know to add tags to the object before publishing? Is there a particular class we need to return instead of 0?

Thank you.

**Comment by mrpdean** [comment, marked as answer]
This is what I do in my Document listener to display a custom message if a user tries to publish a page without a meta description.

Not sure, but I'm assuming it's similar for DataObject listeners. Might help you anyway.

```
if (!$document->getDescription() && $document->getPublished()) {
  throw new \Pimcore\Model\Element\ValidationException("Please enter a description for this page under the SEO & Settings tab above.");
}
```

**Reply by mlkdev** [reply]
Fantastic! That did the trick! TYVM!
